operator/validator/event_handler.go

```diff
@@ -175,7 +183,8 @@ func (c *controller) handleValidatorAddedEvent(
 	sharesExpectedLength := encryptedKeyLength*operatorCount + pubKeysOffset
 
 	if sharesExpectedLength != len(event.Shares) {
-		return nil, errors.Errorf("%s event shares length is not correct", abiparser.ValidatorAdded)
+		malformed = true
+		return nil, &abiparser.MalformedEventError{Err: errors.Errorf("%s event shares length is not correct", abiparser.ValidatorAdded)}
 	}
```

Comment:
we have to set the err and only then return it
```go
if sharesExpectedLength != len(event.Shares) {
  err = &abiparser.MalformedEventError{
	  Err: errors.Errorf(
		  "%s event shares length is not correct: expected %d, got %d",
		  abiparser.ValidatorAdded,
		  sharesExpectedLength,
		  len(event.Shares),
	  ),
  }
  return nil, err
}
```

Reply:
why?

Reply:
I've noticed that there is no handling for a malformed event after the function [`c.onShareCreate(logger, event)`](https://github.com/bloxapp/ssv/blob/c1b73a0581cde68280778cfe465c7a5ada338d21/operator/validator/event_handler.go#L209) is called. Specifically, it appears that `malformed = true` is not set under these circumstances.

This lack is problematic because if `onShareCreate` returns an error, it's unclear whether it's due to a malformed event or a system error. This could result in the nonce being falsely incremented, which could lead to corrupted data

To better manage this, I suggest using the `errors.As()` function as I outlined in [(section 3)](https://github.com/bloxapp/ssv/pull/975#pullrequestreview-1452505021). This function can be used to check if an error is of a specific type, in this case, a malformed event error. By implementing this change, we can better differentiate between error types and prevent false nonce increments.